sound/soc/intel/boards/sof_es8336.c

```diff
@@ -483,6 +505,9 @@ static int sof_es8336_probe(struct platform_device *pdev)
 	}
 	log_quirks(dev);
 
+	if (quirk & SOF_ES8336_ENABLE_DMIC)
+		dmic_be_num = 2;
+
```

Comment:
In the original code, dmic_be_num is set before quirk_override. This mean the quirk_override won't impact on the dmic setting. Now dmic setting will be changed if quirk_override is set. Personally, I like this patch behavior more if there are not products impacted.

Reply:
yes it's intentional that the quirk override will trump NHTL information and DMI.
Of course people could screw up by adding mics in the HDA quirk and removing them in the machine driver quirk. That would lead to an inconsistency between machine driver and topology, but that's an enforced error.